infra/modules/service/main.tf

```diff
@@ -0,0 +1,321 @@
+  # checkov:skip=CKV_AWS_91:Add access logs in future PR
+}
+
+# NOTE: for the demo we expose private http endpoint
+# due to the complexity of acquiring a valid TLS/SSL cert.
+# In a production system we would provision an https listener
+resource "aws_lb_listener" "alb_listener_http" {
+  # TODO(https://github.com/navapbc/template-infra/issues/163) Use HTTPS protocol
+  # checkov:skip=CKV_AWS_2:Implement HTTPS in issue #163
+  # checkov:skip=CKV_AWS_103:Require TLS 1.2 as part of implementing HTTPS support
+
+  load_balancer_arn = aws_lb.alb.arn
+  port              = "80"
+  protocol          = "HTTP"
+
+  default_action {
+    type = "fixed-response"
+
+    fixed_response {
+      content_type = "text/plain"
+      message_body = "Not Found"
+      status_code  = "404"
+    }
+  }
+}
+
+resource "aws_lb_listener_rule" "api_http_forward" {
+  listener_arn = aws_lb_listener.alb_listener_http.arn
+  priority     = 100
+
+  action {
+    type             = "forward"
+    target_group_arn = aws_lb_target_group.api_tg.arn
+  }
+  condition {
+    path_pattern {
+      values = ["/*"]
+    }
+  }
+}
+
+
+resource "aws_lb_target_group" "api_tg" {
+  # you must use a prefix, to facilitate successful tg changes
+  name_prefix          = "api-"
+  port                 = "8080"
+  protocol             = "HTTP"
+  vpc_id               = var.vpc_id
+  target_type          = "ip"
+  deregistration_delay = "30"
+
+  health_check {
+    path                = "/health"
+    port                = 8080
+    healthy_threshold   = 2
+    unhealthy_threshold = 10
+    interval            = 30
+    timeout             = 29
+    matcher             = "200-299"
+  }
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+#######################
+## Service Execution ##
+#######################
+
+resource "aws_ecs_service" "app" {
+  name            = var.service_name
+  cluster         = aws_ecs_cluster.cluster.arn
+  launch_type     = "FARGATE"
+  task_definition = aws_ecs_task_definition.app.arn
+  desired_count   = var.desired_instance_count
+
+  # Allow changes to the desired_count without differences in terraform plan.
+  # This allows autoscaling to manage the desired count for us.
+  lifecycle {
+    ignore_changes = [desired_count]
+  }
+
+  network_configuration {
+    # TODO(https://github.com/navapbc/template-infra/issues/152) set assign_public_ip = false after using private subnets
+    assign_public_ip = true
+    subnets          = var.subnet_ids
+    security_groups  = [aws_security_group.app.id]
+  }
+
+  load_balancer {
+    target_group_arn = aws_lb_target_group.api_tg.arn
+    container_name   = var.service_name
+    container_port   = var.container_port
+  }
+}
+
+resource "aws_ecs_task_definition" "app" {
+  family             = var.service_name
+  execution_role_arn = aws_iam_role.task_executor.arn
+
+  # when is this needed?
+  # task_role_arn      = aws_iam_role.api_service.arn
```

Comment:
typically when the task needs permissions to access other services. e.g. IAM roles allowing S3 access to a task that outputs files to a bucket

Reply:
ah i see, in that case i'll remove it for now since the task doesn't currently access any aws resources